### PR TITLE
Staging 12aug2024

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: include/battle/battle.h [hunk 1]
```diff
@@ -178,13 +179,14 @@ typedef struct UnkBattlemonSub {
...
-    u32 meFirstFlag         : 1;
-    u32 unk4_2F             : 1; // unused
+    u32 knockOffFlag        : 1; // unclear whether true mean knocked off or not knocked
+                                 // off based on current information on its usage
+    u32 metronomeTurns  : 4;     // refers to the item, not the move
+    u32 micleBerryFlag  : 1;
+    u32 custapBerryFlag : 1;
+    u32 quickClawFlag   : 1;
+    u32 meFirstFlag     : 1;
+    u32 unk4_2F         : 1; // unused
```
Comment (PikalaxALT) [inline include/battle/battle.h:189]: somehow splitting this comment across multiple lines broke bitfield alignment

File: include/battle/battle.h [hunk 2]
```diff
@@ -623,11 +626,8 @@ struct BattleSystem {
     u16 unk2454[4];
     u16 unk245C[4];
     int unk2464[4];
-    u32 isRecordingPaused : 1,
-        unk2474_1         : 1,
-        unk2474_2         : 1,
-        unk2474_3         : 1,
-        unk2474_4         : 28;
+    u32 isRecordingPaused : 1, unk2474_1 : 1, unk2474_2 : 1, unk2474_3 : 1,
```
Comment (PikalaxALT) [inline include/battle/battle.h:629]: hey clang-format, you okay?

File: src/options_app.c [hunk 3]
```diff
@@ -801,231 +804,25 @@ static void ov54_021E6418(OptionsApp_Data *data, u16 menuEntryId) {
...
+            color = notSelectedColor;
+        }
+        // required to match a double `bls` above
+        if (i == data->menuEntries[menuEntryId].numStrings - 1) {
+            frameDelay = TEXT_SPEED_NOTRANSFER;
+        } else {
+            frameDelay = TEXT_SPEED_NOTRANSFER;
+        }
+        x = sOptionChoiceLabelXCoords[menuEntryId][i] - (FontID_String_GetWidth(0, data->menuEntries[menuEntryId].strings[i], 0) / 2);
+        AddTextPrinterParameterizedWithColor(&data->windows.selectedOption, 0, data->menuEntries[menuEntryId].strings[i], x, menuEntryId * 24 + 5, frameDelay, color, NULL);
```
Comment (PikalaxALT) [inline src/options_app.c:821]: Compiler's dumb, compiler's really dumb, compiler's SUPER STUPID <:keizaGasm:621346866982027347>

File: include/battle/battle.h (hunk 1 above)
Comment (adrienntindall) [reply include/battle/battle.h:189]: clang might just check sequential bitfield alignment tbh

File: include/battle/battle.h (hunk 2 above)
Comment (adrienntindall) [reply include/battle/battle.h:629]: clearly not but we should probably put a u32 in front of all of these anyways

File: include/constants/bugfix.h [hunk 4]
```diff
@@ -1,7 +1,14 @@
...
-// #define BUGFIX_LOTO_NUMBER_HI
+// When setting the loto number ID, the high 16 bits are set to overwrite the
+// low 16 bits, and the variable that's supposed to hold the high 16 bits is
+// never written. Uncomment this to use the intended behavior. #define
+// BUGFIX_LOTO_NUMBER_HI
+
+// When checking your party against battle rules, the flag that bans Soul Dew
+// uses the most significant bit of the total party level, however this is
+// incorrectly checked. Uncomment this to use the intended behavior. #define
+// BUGFIX_SOUL_DEW_BAN
```
Comment (adrienntindall) [inline include/constants/bugfix.h:12]: Clang fucked with the comments here bad

File: include/field_system.h [hunk 5]
```diff
@@ -143,7 +144,7 @@ struct FieldSystem {
     u8 filler_98[0x4];
     void *unk9C;
     UnkStruct_Fsys_A0 *unkA0;
-    void *unkA4;
+    Save_LinkBattleRuleset *unkA4;
```
Comment (adrienntindall) [inline include/field_system.h:147]: unkA4 -> linkBattleResult

File: src/battle/battle_setup.c [hunk 6]
```diff
@@ -307,58 +297,58 @@ void BattleSetup_InitForFixedLevelFacility(BattleSetup *setup, FieldSystem *fiel
     BattleSetup_SetChatotVoiceClip(setup, chatot, BATTLER_PLAYER);
     setup->storagePC = SaveArray_PCStorage_Get(fieldSystem->saveData);
     setup->timeOfDay = Field_GetTimeOfDay(fieldSystem);
-    setup->unk_10C   = fieldSystem->bagCursor;
+    setup->bagCursor = fieldSystem->bagCursor;
     setup->unk1B8    = fieldSystem->unkB0;
-    setup->unk_12C   = sub_0202CA44(fieldSystem->saveData);
+    setup->unk_12C   = Save_WiFiHistory_Get(fieldSystem->saveData);
```
Comment (adrienntindall) [inline src/battle/battle_setup.c:302]: unk_12C -> wifiHistory

File: src/battle/battle_setup.c [hunk 7]
```diff
@@ -307,58 +297,58 @@ void BattleSetup_InitForFixedLevelFacility(BattleSetup *setup, FieldSystem *fiel
...
-    pokedex           = Save_Pokedex_Get(fieldSystem->saveData);
-    chatot            = Save_Chatot_Get(fieldSystem->saveData);
-    options           = Save_PlayerData_GetOptionsAddr(fieldSystem->saveData);
-    fieldSystem_unkA4 = fieldSystem->unkA4;
+    profile = Save_PlayerData_GetProfileAddr(fieldSystem->saveData);
+    bag     = Save_Bag_Get(fieldSystem->saveData);
+    pokedex = Save_Pokedex_Get(fieldSystem->saveData);
+    chatot  = Save_Chatot_Get(fieldSystem->saveData);
+    options = Save_PlayerData_GetOptionsAddr(fieldSystem->saveData);
+    ruleset = fieldSystem->unkA4;
```
Comment (adrienntindall) [inline src/battle/battle_setup.c:323]: unkA4 -> ruleset

File: src/launch_application.c [hunk 8]
```diff
@@ -869,7 +869,7 @@ static void InitWirelessTradeSelectMonArgs(WirelessTradeSelectMonArgs *args, Fie
     args->profile        = Save_PlayerData_GetProfileAddr(fieldSystem->saveData);
     args->party          = SaveArray_Party_Get(fieldSystem->saveData);
     args->palPad         = SaveArray_Get(fieldSystem->saveData, SAVE_PALPAD);
-    args->unk14          = sub_0202CA44(fieldSystem->saveData);
+    args->unk14          = Save_WiFiHistory_Get(fieldSystem->saveData);
```
Comment (adrienntindall) [inline src/launch_application.c:872]: unk14 -> wifiHistory

File: src/launch_application.c [hunk 9]
```diff
@@ -1179,7 +1179,7 @@ void sub_0203F844(FieldSystem *fieldSystem, u16 a1) {
     args->pcStorage          = SaveArray_PCStorage_Get(fieldSystem->saveData);
     args->pokedex            = Save_Pokedex_Get(fieldSystem->saveData);
     args->unk14              = sub_0202C6F4(fieldSystem->saveData);
-    args->unk18              = sub_0202CA44(fieldSystem->saveData);
+    args->unk18              = Save_WiFiHistory_Get(fieldSystem->saveData);
```
Comment (adrienntindall) [inline src/launch_application.c:1182]: unk18 -> wifiHistory

File: src/map_preview_graphic.c [hunk 10]
```diff
@@ -54,143 +54,706 @@ typedef struct UnkStruct_0206A388 {
...
+                  .palNo  = NARC_preview_graphic_preview_graphic_dark_cave_r31_night_NCLR,
+                  .charNo = NARC_preview_graphic_preview_graphic_dark_cave_r31_night_NCGR_lz,
+                  .scrnNo = NARC_preview_graphic_preview_graphic_dark_cave_r31_night_NSCR_lz,
+            },
+        },
+     .unused3C = 0,
+     },
+    {
+     // Sprout Tower
+        .mapId   = MAP_D15R0101,
```
Comment (adrienntindall) [inline src/map_preview_graphic.c:115]: not sur why but .mapId's are unaligned here, any way to fix it?

File: src/options_app.c (hunk 3 above)
Comment (adrienntindall) [reply src/options_app.c:821]: what

File: src/save_wifi_history.c [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+    SaveSubstruct_UpdateCRC(SAVE_WIFI_HISTORY);
+}
+
+SaveWiFiHistory *Save_WiFiHistory_Get(SaveData *saveData) {
+    SaveSubstruct_AssertCRC(SAVE_WIFI_HISTORY);
+    return (SaveWiFiHistory *)SaveArray_Get(saveData, SAVE_WIFI_HISTORY);
+}
+
+void WiFiHistory_SetPlayerGlobeInfo(SaveWiFiHistory *wifiHistory, int country, int region) {
+    GF_ASSERT(country < 256);
```
Comment (adrienntindall) [inline src/save_wifi_history.c:30]: COUNTRY_MAX

File: src/save_wifi_history.c [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+    GF_ASSERT(region < 64);
```
Comment (adrienntindall) [inline src/save_wifi_history.c:31]: REGION_MAX

**Review by adrienntindall — CHANGES_REQUESTED** [review]
Changes were pushed since I started my review so I'll update with more in a second

File: src/map_preview_graphic.c (hunk 10 above)
Comment (PikalaxALT) [reply src/map_preview_graphic.c:115]: > not sur why but .mapId's are unaligned here, any way to fix it?

yes: revert .clang-format

File: include/battle/battle_setup.h [hunk 13]
```diff
@@ -37,7 +37,7 @@ struct BattleSetup {                     // declared in trainer_data.h
     PCStorage *storagePC;                // 114
     SOUND_CHATOT *chatot[BATTLER_MAX];   // 118
     void *unk_128;
-    void *unk_12C;
+    void *wifiHistory;
```
Comment (adrienntindall) [inline include/battle/battle_setup.h:40]: void -> WifiHistory or whatever the struct was

**Review by adrienntindall — CHANGES_REQUESTED** [review]

**Review by adrienntindall — APPROVED** [review]

**Pull request merged**